test/decl/protocol/special/coding/struct_codable_simple.swift

```diff
@@ -29,3 +29,21 @@ let _ = SimpleStruct.encode(to:)
 // The synthesized CodingKeys type should not be accessible from outside the
 // struct.
 let _ = SimpleStruct.CodingKeys.self // expected-error {{'CodingKeys' is inaccessible due to 'private' protection level}}
+
+// rdar://problem/59655704 
+struct SR_12248_1: Codable { // expected-error {{type 'SR_12248_1' does not conform to protocol 'Encodable'}}
+  var x: Int // expected-note {{'x' previously declared here}}
+  var x: Int // expected-error {{invalid redeclaration of 'x'}}
+  // expected-note@-1 {{cannot automatically synthesize 'Encodable' because '<<error type>>' does not conform to 'Encodable'}}
```

Comment:
Not sure if this is a requirement for the SR, but I think this shouldn't be noted when the synthesize type `hasError()`, in fact, the missing `Encodable` conformance diagnostic itself shouldn't be diagnosed in this case.
WDYT?

Reply:
Honestly, I also thought of the same thing. I don’t think this needs to be tested. However, as a newbie I decided to add a lot of tests 😅

Reply:
You could take a look at `DerivedConformance::deriveEncodable`, note that in 

https://github.com/apple/swift/blob/426583523e063be1ed75027044f56873b7b73183/lib/Sema/TypeCheckDeclPrimary.cpp#L721 

The member is flagged as invalid, so on `DerivedConformance::deriveEncodable` and decodable you could check for any nominal member and skip the conformance diagnostic if any of the members are invalid :)

Reply:
The `ErrorType`s are unfortunate, but it's not enough to block the PR.